lib/logstash/codecs/cef.rb

```diff
@@ -348,51 +348,25 @@ def encode(event)
   # Escape pipes and backslashes in the header. Equal signs are ok.
   # Newlines are forbidden.
   def sanitize_header_field(value)
-    output = String.new
-
-    value = value.to_s.gsub(/\r\n/, "\n")
-
-    value.each_char{|c|
-      case c
-      when "\\", "|"
-        output << "\\#{c}"
-      when "\n", "\r"
-        output << " "
-      else
-        output << c
-      end
-    }
-
-    return output
+    value.to_s
+         .gsub(/\r\n/, "\n")
+         .gsub(/[\\\|\n\r]/, {"\\" => "\\\\", "|" => "\\|", "\n" => " ", "\r" => " "})
   end
 
   # Keys must be made up of a single word, with no spaces
   # must be alphanumeric
   def sanitize_extension_key(value)
-    value = value.to_s.gsub(/[^a-zA-Z0-9]/, "")
-    return value
+    value.to_s
+         .gsub(/[^a-zA-Z0-9]/, "")
   end
 
   # Escape equal signs in the extensions. Canonicalize newlines.
   # CEF spec leaves it up to us to choose \r or \n for newline.
   # We choose \n as the default.
   def sanitize_extension_val(value)
-    output = String.new
-
-    value = value.to_s.gsub(/\r\n/, "\n")
-
-    value.each_char{|c|
-      case c
-      when "\\", "="
-        output << "\\#{c}"
-      when "\n", "\r"
-        output << "\\n"
-      else
-        output << c
-      end
-    }
-
-    return output
+    value.to_s
+         .gsub(/\r\n/, "\n")
+         .gsub(/[\\=\n\r]/, {"\\" => "\\\\", "=" => "\\=", "\n" => "\\n", "\r" => "\\n"})
```

Comment:
these hashes could be static, probably won't matter much though ...

Reply:
I'll see if it changes perf much.

Reply:
Excellent tip. It looks like we get about 10% better with the gsub args constantized.

Reply:
"Shame" on JRuby possibly missing an optimization :baby_bottle: :fast_forward: